internal/core/engine/engine_test.go

```
@@ -0,0 +1,419 @@
+}
+
+func TestEngine_DeleteSession(t *testing.T) {
+	engine, commands, events := createTestEngine(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Start engine
+	err := engine.Start(ctx)
+	require.NoError(t, err)
+	defer engine.Stop()
```

golangci-lint (1.64.8)

Error return value of `engine.Stop` is not checked

(errcheck)